init.sh

```diff
@@ -1,44 +1,50 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env node
+'use strict';
 
-def cp(src, dest, app_name)
-  if File.directory?(src)
-    Dir.mkdir(dest) unless Dir.exists?(dest)
-  else
-    content = File.read(src)
-      .gsub("SampleApp", app_name)
-      .gsub("Examples/#{app_name}/", "")
-      .gsub("../../Libraries/", "node_modules/react-native/Libraries/")
-      .gsub("../../React/", "node_modules/react-native/React/")
-    File.write(dest, content)
-  end
-end
+var path = require('path');
+var fs = require('fs');
+var file = require('file');
 
-def main(dest, app_name)
-  source = File.expand_path("../Examples/SampleApp", __FILE__)
-  files = Dir.chdir(source) { Dir["**/*"] }
-    .reject { |file| file["project.xcworkspace"] || file["xcuserdata"] }
-    .each { |file|
-      new_file = file.gsub("SampleApp", app_name)
-      cp File.join(source, file), File.join(dest, new_file), app_name
-    }
-end
+if (process.argv.length === 0) {
+  console.log('Usage: ' + path.basename(__filename) + ' <ProjectNameInCamelCase>');
```

Comment:
Not 100% on this, but from when I started this over the weekend as well, I think that ruby’s `__FILE__` returns just the filename (`init.sh`), while node’s `__filename` includes the full path resolution (`/Users/user/path/to/react-native/init.sh`).